engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -205,19 +212,88 @@ private class Factory(
+        val hydrationArgs = getHydrationArguments(hydration, parentType, hydratedFieldDef, actorFieldDef)
+
+        val matchStrategy = if (hydration.isObjectMatchByIndex) {
+            NadelBatchHydrationMatchStrategy.MatchIndex
+        } else if (hydration.objectIdentifier.contains(",")) {
+            NadelBatchHydrationMatchStrategy.MatchObjectIdentifiers(
+                hydration.objectIdentifier.split(",")
+                    .map { objectIdentifier ->
+                        // Todo switch off using colon and use a complex GraphQL object or something rather
+                        NadelBatchHydrationMatchStrategy.MatchObjectIdentifier(
+                            sourceId = objectIdentifier.substringBefore(":", missingDelimiterValue = "")
+                                .takeIf {
+                                    it.isNotEmpty()
+                                }
+                                ?.let {
+                                    NadelQueryPath(segments = it.split("."))
+                                }
+                                ?: hydrationArgs
+                                    .asSequence()
+                                    .map(NadelHydrationActorInputDef::valueSource)
+                                    .filterIsInstance<FieldResultValue>()
+                                    .single()
+                                    .queryPathToField,
+                            resultId = objectIdentifier.substringAfter(":"),
+                        )
+                    },
+            )
+        } else {
+            NadelBatchHydrationMatchStrategy.MatchObjectIdentifier(
+                sourceId = hydrationArgs
+                    .asSequence()
+                    .map(NadelHydrationActorInputDef::valueSource)
+                    .filterIsInstance<FieldResultValue>()
+                    .single()
+                    .queryPathToField,
+                resultId = hydration.objectIdentifier.substringAfter(":"),
+            )
+        }
+
         return NadelBatchHydrationFieldInstruction(
             location = location,
             hydratedFieldDef = hydratedFieldDef,
             actorService = actorService,
             queryPathToActorField = NadelQueryPath(listOfNotNull(hydration.syntheticField, hydration.topLevelField)),
             actorFieldDef = actorFieldDef,
-            actorInputValueDefs = getHydrationArguments(hydration, parentType, hydratedFieldDef, actorFieldDef),
+            actorInputValueDefs = hydrationArgs,
             timeout = hydration.timeout,
             batchSize = batchSize,
-            batchHydrationMatchStrategy = if (hydration.isObjectMatchByIndex) {
-                NadelBatchHydrationMatchStrategy.MatchIndex
-            } else {
-                NadelBatchHydrationMatchStrategy.MatchObjectIdentifier(objectId = hydration.objectIdentifier)
+            batchHydrationMatchStrategy = matchStrategy,
+            sourceFields = Unit.let {
```

Comment:
Calculates the required `sourceFields` based on `arguments` and the complex `identifiedBy`